pkg/operation/botanist/component/kubeapiserver/deployment.go

```diff
@@ -697,6 +703,41 @@ func (k *kubeAPIServer) handleSNISettings(deployment *appsv1.Deployment) {
 	deployment.Spec.Template.Spec.Containers[0].Command = append(deployment.Spec.Template.Spec.Containers[0].Command, fmt.Sprintf("--advertise-address=%s", k.values.SNI.AdvertiseAddress))
 }
 
+func (k *kubeAPIServer) handleTLSSNISettings(deployment *appsv1.Deployment, secretNames []string) error {
+	if len(k.values.SNI.TLS) != len(secretNames) {
+		return fmt.Errorf("number of TLS SNI configs and provided secret names is not equal (%d vs. %d)", len(k.values.SNI.TLS), len(secretNames))
+	}
```

Comment:
I don't think it can happen right now - this is just a "safety net" to not panic since we access the slice with indexes. Who knows how this code will be refactored in the future, maybe it will safe us one day 😉